### ISSUE TITLE
Input with physical keyboard (BlackBerry Priv)

### ISSUE THREAD

**Issue by duckduck1** [signalapp/Signal-Android#6495]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
Editing a text at cursors location stops working with BlackBerry Privs physical keyboard. This bug occurs from time to time.

### Steps to reproduce
- use BlackBerry Privs physical keyboard
- write a text in conversation
- move cursor to another location
- press backspace to delete text
- Repeat above steps to force the bug
- unexpected behaviour occurs

**Actual result:** Cursor jumps to the end of the text.
**Expected result:** Cursor should stay in position and delete character inbetween the text.

### Device info
**Device:** BlackBerry Priv
**Android version:** 6.0.1
**Signal version:** 4.1.0

### Link to debug log
https://gist.github.com/anonymous/16c1d8c3807be10cb6db77b03ef7be05

https://gist.github.com/anonymous/af4c932ce00f06939a74248282ce9d31 (old log)

**Comment by duckduck1** [comment]
Here is another bug report that even resulted in a force close.

It starts with a FC of the same situation on a past date. Current FC happened in the end with last time stamp. It is messy due to errors, sorry. I did not want to remove anything important.

**Comment by duckduck1** [comment]
Forgot about the link to bug report: https://gist.github.com/anonymous/38ea24c06e543f332c3aa8fb5ea271f0

**Comment by toolgirl** [comment]
I occasionally have the same issue but not on a Blackberry physical keyboard. Mine is a Nexus 5 running Android 6.0.1. 

This bug describes my issue. I can't seem to force it anymore though. It used to happen anytime I wanted to insert anything into the middle of a message text. Whereas now it only happens very occasionally. It used to crash and I submitted bug-reports through the phone but didn't know about doing it through this platform.
It used to be more common but it happens less frequently which is why it is harder to capture.

Here is a link to the bug-report I captured:
https://gist.github.com/527af08291611938666551f2f5c1a37a
